casepro/backend/junebug.py

```diff
@@ -0,0 +1,522 @@
+from datetime import datetime
+from django.conf import settings
+from django.conf.urls import url
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+import json
+import requests
+import pytz
+
+from . import BaseBackend
+from ..contacts.models import Contact
+from ..msgs.models import Message
+from ..utils import uuid_to_int
+
+from dash.utils import is_dict_equal
+from dash.utils.sync import BaseSyncer, sync_local_to_changes
+
+from itertools import chain
+
+
+class IdentityStore(object):
```

Comment:
At the moment we fetch the URN from the identity store when we need it (when we send an outgoing message). It might lead to a small performance increase, but I don't see a strong argument for changing the way things work now.

If there's a good argument for syncing the URN when we sync the contact, and using that instead for sending outbound messages, then we can change things, but I don't think it's quite worth the effort at the moment, and as far as I know we have no need for surfacing it in the UI. (It might also introduce problems if a contact's URN gets changed in between syncs, and we send the message to the old URN).